contrib/pam_zfs_key/pam_zfs_key.c

```diff
@@ -368,7 +368,7 @@ change_key(pam_handle_t *pamh, const char *ds_name,
 
 static int
 decrypt_mount(pam_handle_t *pamh, const char *ds_name,
-    const char *passphrase)
+    const char *passphrase, int do_mount)
```

Comment:
Instead of adding bool flag here I'd suggest moving mount functionality to a separate function. `decrypt_mount()` which doesn't mount looks very weird to me. This function should be named `load_encryption_key()` or something.